### PR TITLE
Central components (AIO Installation) Windows and  MacOS tests integration

### PR DESCRIPTION
**Pull request by pro-akim**
## Description

Integration between Windows, MacOS agent test and Central Component tests

### CODE REVIEW

**Comment by pro-akim** [comment]
## Update

<details><summary>Central</summary>

```
version: 0.1
description: This workflow is used to test manager deployment for DDT1 PoC
variables:
  manager-os:
    - linux-ubuntu-20.04-amd64
    - linux-ubuntu-22.04-amd64
    - linux-oracle-9-amd64
    - linux-amazon-2-amd64
    - linux-redhat-7-amd64
    - linux-redhat-8-amd64
    - linux-redhat-9-amd64
    - linux-centos-7-amd64
    - linux-centos-8-amd64
    - linux-debian-10-amd64
    - linux-debian-11-amd64
    - linux-debian-12-amd64
  infra-provider: vagrant
  working-dir: /tmp/dtt1-poc

tasks:
  # Unique manager allocate task
  - task: "allocate-manager-{manager}"
    description: "Allocate resources for the manager."
    do:
      this: process
      with:
        path: python3
        args:
          - modules/allocation/main.py
          - action: create
          - provider: "{infra-provider}"
          - size: large
          - composite-name: "{manager}"
          - inventory-output: "{working-dir}/manager-{manager}/inventory.yaml"
          - track-output: "{working-dir}/manager-{manager}/track.yaml"
    on-error: "abort-all"
    foreach:
      - variable: manager-os
        as: manager

  # Generic manager test task
  - task: "run-manager-tests"
    description: "Run tests install for the manager."
    do:
      this: process
      with:
        path: python3
        args:
          - modules/testing/main.py
          - targets:
            - wazuh-1: "{working-dir}/manager-linux-centos-7-amd64/inventory.yaml"
            - wazuh-2: "{working-dir}/manager-linux-ubuntu-20.04-amd64/inventory.yaml"
            - wazuh-3: "{working-dir}/manager-linux-ubuntu-22.04-amd64/inventory.yaml"
            - wazuh-4: "{working-dir}/manager-linux-oracle-9-amd64/inventory.yaml"
            - wazuh-5: "{working-dir}/manager-linux-amazon-2-amd64/inventory.yaml"
            - wazuh-6: "{working-dir}/manager-linux-redhat-7-amd64/inventory.yaml"
            - wazuh-7: "{working-dir}/manager-linux-redhat-8-amd64/inventory.yaml"
            - wazuh-8: "{working-dir}/manager-linux-redhat-9-amd64/inventory.yaml"
            - wazuh-9: "{working-dir}/manager-linux-centos-8-amd64/inventory.yaml"
            - wazuh-10: "{working-dir}/manager-linux-debian-10-amd64/inventory.yaml"
            - wazuh-11: "{working-dir}/manager-linux-debian-11-amd64/inventory.yaml"
            - wazuh-12: "{working-dir}/manager-linux-debian-12-amd64/inventory.yaml"
          - tests: "install,restart,stop,uninstall"
          - component: "manager"
          - wazuh-version: "4.7.4"
          - wazuh-revision: "40717"
          - live: "True"


```
</details>

[workflow.log](https://github.com/wazuh/wazuh-qa/files/15197905/workflow.log)

**Comment by pro-akim** [comment]
## Status

||AWS|Vagrant|
|--|--|--|
|agent linux|:green_circle: |  |
|agent macos|   | mac stadium in use  |
|agent windows|:green_circle:  |  |
|central components|  | :green_circle:   |
|manager|  | :green_circle:  |

**Comment by pro-akim** [comment]
## Status     
    
||AWS|Vagrant|
|--|--|--|
|Wazuh agent linux|:green_circle: |:green_circle:  |
|Wazuh agent macos| :green_circle:   | :green_circle:   |
|Wazuh agent windows|:green_circle:  |  :red_circle:   |
|Wazuh central components| :green_circle:   | :green_circle:   |
|Wazuh manager| :green_circle:  | :green_circle:  |

File: deployability/modules/testing/tests/test_agent/test_install.py [hunk 1]
```diff
@@ -82,7 +82,9 @@ def test_installation(wazuh_params):
 
     # Agent installation
     for agent_name, agent_params in wazuh_params['agents'].items():
-        WazuhAgent.perform_install_and_scan_for_agent(agent_params, agent_name, wazuh_params)
+        #WazuhAgent.perform_install_and_scan_for_agent(agent_params, agent_name, wazuh_params)
```
Comment (rauldpm) [inline deployability/modules/testing/tests/test_agent/test_install.py:85]: Remove comment

File: deployability/modules/testing/tests/test_agent/test_uninstall.py [hunk 2]
```diff
@@ -79,8 +79,8 @@ def test_uninstall(wazuh_params):
 
     # Agent uninstallation
     for agent_names, agent_params in wazuh_params['agents'].items():
-        WazuhAgent.perform_uninstall_and_scan_for_agent(agent_params,wazuh_params)
-
+        #WazuhAgent.perform_uninstall_and_scan_for_agent(agent_params,wazuh_params)
```
Comment (rauldpm) [inline deployability/modules/testing/tests/test_agent/test_uninstall.py:82]: Remove comment

File: deployability/modules/testing/tests/test_central_components/test_install.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    for _, manager_params in wazuh_params['managers'].items():
+        Utils.check_inventory_connection(manager_params)
+        HostConfiguration.disable_firewall(manager_params)
+
+    # Certs create and scp from master to worker
+    HostConfiguration.certs_create(wazuh_params['wazuh_version'], wazuh_params['master'], wazuh_params['dashboard'], wazuh_params['indexers'], wazuh_params['workers'])
+
+    # Install central components and perform checkfile testing
+    for _, manager_params in wazuh_params['managers'].items():
+        #WazuhCentralComponents.perform_install_and_scan_for_aio(manager_params, wazuh_params)
```
Comment (rauldpm) [inline deployability/modules/testing/tests/test_central_components/test_install.py:65]: Remove comment

File: deployability/modules/testing/tests/test_central_components/test_uninstall.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+
+def test_uninstall(wazuh_params):
+    assert 'active' in GeneralComponentActions.get_component_status(wazuh_params['master'], 'wazuh-manager'), logger.error(f'The manager in {HostInformation.get_os_name_and_version_from_inventory(wazuh_params["master"])} is not active')
+    assert 'active' in GeneralComponentActions.get_component_status(wazuh_params['dashboard'], 'wazuh-dashboard'), logger.error(f'The dashboard in {HostInformation.get_os_name_and_version_from_inventory(wazuh_params["dashboard"])} is not active')
+    for indexer_params in wazuh_params['indexers']:
+        assert 'active' in GeneralComponentActions.get_component_status(indexer_params, 'wazuh-indexer'), logger.error(f'The indexer in {HostInformation.get_os_name_and_version_from_inventory(indexer_params)} is not active')
+    assert 'active' in GeneralComponentActions.get_component_status(wazuh_params['master'], 'filebeat'), logger.error(f'The filebeat in {HostInformation.get_os_name_and_version_from_inventory(wazuh_params["master"])} is not active')
+
+    #WazuhCentralComponents.perform_uninstall_and_scan_for_aio(wazuh_params['master'])
```
Comment (rauldpm) [inline deployability/modules/testing/tests/test_central_components/test_uninstall.py:59]: Remove comment

File: deployability/modules/testing/tests/test_manager/test_install.py [hunk 5]
```diff
@@ -61,7 +61,8 @@ def test_installation(wazuh_params):
 
     # Install managers and perform checkfile testing
     for manager_name, manager_params in wazuh_params['managers'].items():
-        WazuhManager.perform_install_and_scan_for_manager(manager_params, manager_name, wazuh_params)
+        #WazuhManager.perform_install_and_scan_for_manager(manager_params, manager_name, wazuh_params)
```
Comment (rauldpm) [inline deployability/modules/testing/tests/test_manager/test_install.py:64]: Remove comment

File: deployability/modules/testing/tests/test_manager/test_uninstall.py [hunk 6]
```diff
@@ -51,8 +52,8 @@ def test_uninstall(wazuh_params):
         manager_status = GeneralComponentActions.get_component_status(manager, 'wazuh-manager')
         assert 'active' in manager_status, logger.error(f'The {HostInformation.get_os_name_and_version_from_inventory(manager)} is not active')
     for _, manager_params in wazuh_params['managers'].items():
-        WazuhManager.perform_uninstall_and_scan_for_manager(manager_params)
-
+        #WazuhManager.perform_uninstall_and_scan_for_manager(manager_params)
```
Comment (rauldpm) [inline deployability/modules/testing/tests/test_manager/test_uninstall.py:55]: Remove comment

**Review by rauldpm — CHANGES_REQUESTED** [review]

File: deployability/modules/testing/tests/test_manager/test_uninstall.py (hunk 6 above)
Comment (pro-akim) [reply deployability/modules/testing/tests/test_manager/test_uninstall.py]: Done in https://github.com/wazuh/wazuh-qa/pull/5324/commits/96c47257f1bca33c7f144285a4c7b63a8c10fbce

File: deployability/modules/testing/tests/test_manager/test_install.py (hunk 5 above)
Comment (pro-akim) [reply deployability/modules/testing/tests/test_manager/test_install.py]: Done in https://github.com/wazuh/wazuh-qa/pull/5324/commits/96c47257f1bca33c7f144285a4c7b63a8c10fbce

File: deployability/modules/testing/tests/test_central_components/test_uninstall.py (hunk 4 above)
Comment (pro-akim) [reply deployability/modules/testing/tests/test_central_components/test_uninstall.py]: Done in https://github.com/wazuh/wazuh-qa/pull/5324/commits/96c47257f1bca33c7f144285a4c7b63a8c10fbce

File: deployability/modules/testing/tests/test_central_components/test_install.py (hunk 3 above)
Comment (pro-akim) [reply deployability/modules/testing/tests/test_central_components/test_install.py]: Done in https://github.com/wazuh/wazuh-qa/pull/5324/commits/96c47257f1bca33c7f144285a4c7b63a8c10fbce

File: deployability/modules/testing/tests/test_agent/test_uninstall.py (hunk 2 above)
Comment (pro-akim) [reply deployability/modules/testing/tests/test_agent/test_uninstall.py]: Done in https://github.com/wazuh/wazuh-qa/pull/5324/commits/96c47257f1bca33c7f144285a4c7b63a8c10fbce

File: deployability/modules/testing/tests/test_agent/test_install.py (hunk 1 above)
Comment (pro-akim) [reply deployability/modules/testing/tests/test_agent/test_install.py]: Done in https://github.com/wazuh/wazuh-qa/pull/5324/commits/96c47257f1bca33c7f144285a4c7b63a8c10fbce

**Review by rauldpm — APPROVED** [review]

**Pull request merged**